### PR TITLE
[PySpark] Add approxCountDistinct, add_months, and various array functions

### PR DESCRIPTION
**Pull request by binste**
Adds `approxCountDistinct`, `add_months`, and all remaining array functions except for `array_compact` and `array_remove` which depend on https://github.com/duckdb/duckdb/discussions/14696.

### CODE REVIEW

**Comment by Mytherin** [comment]
Thanks - could you just have a look at the merge conflicts?

**Comment by binste** [comment]
Done :)

**Comment by binste** [comment]
Just fyi, I've added array_compact and array_remove in #14761 so this PR is ready whenever you have a moment to review. No rush :)

File: tools/pythonpkg/duckdb/experimental/spark/sql/functions.py [hunk 1]
```diff
@@ -80,6 +81,44 @@ def struct(*cols: Column) -> Column:
...
+    >>> df.select(array([df.age, df.age]).alias("arr")).collect()
+    [Row(arr=[2, 2]), Row(arr=[5, 5])]
+    >>> df.select(array('age', 'age').alias("col")).printSchema()
+    root
+     |-- col: array (nullable = false)
+     |    |-- element: long (containsNull = true)
+    """
+    if len(cols) == 1 and isinstance(cols[0], (list, set)):
+        cols = cols[0]
+    return _invoke_function_over_columns("list_value", *cols)
```
Comment (Tishj) [inline tools/pythonpkg/duckdb/experimental/spark/sql/functions.py:119]: in duckdb there is a difference between array and list, `array` being a fixed size list.
I'm not sure spark has the same distinction? Just thought I'd mention it, that this *might* be confusing depending on what system the user is expecting the behavior of

**Review by Tishj — APPROVED** [review]
Thanks!
Just had one remark about the use of list/array, but other than that everything looks great

**Comment by Mytherin** [comment]
Thanks!

File: tools/pythonpkg/duckdb/experimental/spark/sql/functions.py (hunk 1 above)
Comment (binste) [reply tools/pythonpkg/duckdb/experimental/spark/sql/functions.py:119]: Agree, it's confusing if someone would expect DuckDB behaviour. As the goal is to have compatibility with the PySpark API, I hope that's not the case but something to keep an eye out! In my understanding, Spark's `ArrayType` is what are `list`s in DuckDB and there is no equivalent fixed-size version of it.

**Pull request merged**